src/grafana_dashboards/vault.json

```diff
@@ -858,6 +857,95 @@
             ],
             "title": "Allocated Bytes",
             "type": "stat"
+        },
+        {
+            "collapsed": false,
+            "gridPos": {
+                "h": 1,
+                "w": 24,
+                "x": 0,
+                "y": 28
+            },
+            "id": 96,
+            "panels": [],
+            "title": "Audit Logs",
+            "type": "row"
+        },
+        {
+            "datasource": {
+                "type": "loki",
+                "uid": "PCFE195BA03E127CE"
+            },
+            "description": "",
+            "gridPos": {
+                "h": 8,
+                "w": 11,
+                "x": 0,
+                "y": 29
+            },
+            "id": 93,
+            "links": [],
+            "options": {
+                "dedupStrategy": "none",
+                "enableLogDetails": true,
+                "prettifyLogMessage": false,
+                "showCommonLabels": false,
+                "showLabels": false,
+                "showTime": true,
+                "sortOrder": "Descending",
+                "wrapLogMessage": false
+            },
+            "targets": [
+                {
+                    "datasource": {
+                        "type": "loki",
+                        "uid": "PCFE195BA03E127CE"
+                    },
+                    "editorMode": "code",
+                    "expr": "{charm=\"vault-k8s\"} | json | type = `response` or type = `request`",
```

Comment:
I don't think we need to be explicit about the charm name here. We can use `$juju_application` to represent it.

Reply:
If I use any combination of the `$juju_application`, `$juju_model`, `$juju_model_uuid` or `$juju_unit`, I get an error saying Loki does not support regexes that include the empty string. In fact, we can see that it is included in the other queries:
`sum(vault_secret_kv_count{juju_application=~".*",juju_model=~".*",juju_model_uuid=~".*",juju_unit=~".*"})`

I can either create a new variable that doesn't include the empty string, or change the regexes to be ".+" instead of ".*"

Reply:
also, this regex will match all applications. If there was a different application that provided logs to this loki datasource with a field `type = response or type = request`, it would also match and show up here.

Reply:
This sounds like a bug, you may want to create an issue in the Grafana charm
- https://github.com/canonical/grafana-k8s-operator

Reply:
No, it's meant to be this way 
```
parse error : queries require at least one regexp or equality matcher that does not have an empty-compatible value. For instance, app=~".*" does not meet this requirement, but app=~".+" will
```

https://github.com/grafana/loki/issues/7816

Reply:
I'm not saying loki should accept an empty string. My point is that the $juju_application variable should exist for the loki data source (?)

Reply:
It does exist, it's a query variable with the regex ".*" for the prometheus source. This regex is incompatible with loki. Using ".+" instead matches any non-empty string, which doesn't break the rest of the panels and also makes this variable work properly with loki.

Reply:
I've updated this to remove the regexes from the variables. Everything is working as expected and we still have variables everywhere, let me know what you think.